system/Database/BaseBuilder.php

```diff
@@ -2653,6 +2653,105 @@ public function delete($where = '', ?int $limit = null, bool $resetData = true)
         return $this->testMode ? $sql : $this->db->query($sql, $this->binds, false);
     }
 
+    /**
+     * Sets data and calls batchExecute to run queries
+     *
+     * @param array|object|null $set         a dataset or select query
+     * @param array|RawSql|null $constraints
+     *
+     * @return false|int|string[] Number of rows affected or FALSE on failure, SQL array when testMode
+     */
+    public function deleteBatch($set = null, $constraints = null, int $batchSize = 100)
+    {
+        $this->onConstraint($constraints);
+
+        if ($set !== null && $set !== []) {
+            $this->setData($set, true);
+        }
+
+        return $this->batchExecute('_deleteBatch', $batchSize);
+    }
+
+    /**
+     * Generates a platform-specific batch update string from the supplied data
+     */
+    protected function _deleteBatch(string $table, array $keys, array $values): string
+    {
+        $sql = $this->QBOptions['sql'] ?? '';
+
+        // if this is the first iteration of batch then we need to build skeleton sql
+        if ($sql === '') {
+            $constraints = $this->QBOptions['constraints'] ?? [];
+
+            if ($constraints === []) {
+                if ($this->db->DBDebug) {
+                    throw new DatabaseException('You must specify a constraint to match on for batch deletes.'); // @codeCoverageIgnore
+                }
+
+                return ''; // @codeCoverageIgnore
+            }
+
+            $alias = $this->QBOptions['alias'] ?? '_u';
+
+            $sql = 'DELETE ' . $table . ' FROM ' . $table . "\n";
+
+            $sql .= "INNER JOIN (\n{:_table_:}";
+
+            $sql .= ') ' . $alias . "\n";
+
+            $sql .= 'ON ' . implode(
+                ' AND ',
+                array_map(
+                    static fn ($key, $value) => (
+                        $value instanceof RawSql ?
+                        $value :
+                        (
+                            is_string($key) ?
+                            $table . '.' . $key . ' = ' . $alias . '.' . $value :
+                            $table . '.' . $value . ' = ' . $alias . '.' . $value
+                        )
+                    ),
+                    array_keys($constraints),
+                    $constraints
+                )
+            );
+
+            // convert binds in where
+            foreach ($this->QBWhere as $key => $where) {
+                foreach ($this->binds as $field => $bind) {
+                    $this->QBWhere[$key]['condition'] = str_replace(':' . $field . ':', $bind[0], $where['condition']);
+                }
+            }
+
+            // remove database prefix from alias in where
+            $sql .= ' ' . str_replace(
+                $this->db->DBPrefix . trim($alias, $this->db->escapeChar),
+                trim($alias, $this->db->escapeChar),
+                $this->compileWhereHaving('QBWhere')
+            );
```

Comment:
Can you add a test case for this replacement?

Reply:
This replacement is not longer necessary when alias tracking was added to setAlias(). I have removed the replacement in all these functions. I did add a where() clause to the test case to validate any sql generated.

```php
    public function setAlias(string $alias): BaseBuilder
    {
        if ($alias !== '') {
            $this->db->addTableAlias($alias); 
```